packages/babel-types/scripts/generators/builders.js

```diff
@@ -118,10 +150,13 @@ import type * as t from "../..";
   Object.keys(definitions.DEPRECATED_KEYS).forEach(type => {
     const newType = definitions.DEPRECATED_KEYS[type];
     const formatedBuilderName = formatBuilderName(type);
+    const formatedNewBuilderName = formatBuilderName(newType);
     output += `/** @deprecated */
 function ${type}(${generateBuilderArgs(newType).join(", ")}): t.${type} {
   console.trace("The node type ${type} has been renamed to ${newType}");
-  return builder.apply("${type}", arguments);
+  return ${formatedNewBuilderName}(${t.BUILDER_KEYS[newType].join(
+      ", "
+    )}) as unknown as t.${type};
```

Comment:
The type casting here is wrong, the deprecated AST builders are returning the renamed AST nodes.